### PR TITLE
[CP] Fixes #23364 - add search input in breadcrumbs switcher

### PR DESCRIPTION
**Pull request by amirfefer**
Following to this [discussion](https://community.theforeman.org/t/cherry-pick-breadcrumbs-switcher-search-input-to-1-18/9743)
(cherry picked from commit [dc86fdd](https://github.com/theforeman/foreman/commit/dc86fdd8a92f4aed6eb01313ac436bd01ba54869) and [183c8dc](https://github.com/theforeman/foreman/commit/183c8dca929c78efb9c6562f0e7aa61538128a60))

### CODE REVIEW

File: test/integration/breadcrumbs_switcher_test.rb [hunk 1]
```diff
@@ -12,4 +12,15 @@ class BreadcrumbsSwitcherTest < IntegrationTestWithJavascript
       assert_equal current_bookmark_name, 'three'
     end
   end
+
+  test "search query" do
+    visit bookmarks_path
+    click_link 'foo'
+    click_button 'switcher'
+    fill_in('breadcrumbs-search', :with => 'three')
+    wait_for_ajax
```
Comment (amirfefer) [inline test/integration/breadcrumbs_switcher_test.rb:21]: This test depends on this [PR](https://github.com/theforeman/foreman/pull/5541) which fixes the `wait_for_ajax ` **bug** (it didn't work with react).  Could we cp this PR as well?

Comment (mmoll) [reply test/integration/breadcrumbs_switcher_test.rb:21]: @amirfefer please add that commit to this PR

Comment (amirfefer) [reply test/integration/breadcrumbs_switcher_test.rb:21]: @mmoll - done :)

**Review by ohadlevy — APPROVED** [review]

**Review by mmoll — APPROVED** [review]

**Comment by ohadlevy** [comment]
[test katello]

File: package.json [hunk 2]
```diff
@@ -77,6 +77,7 @@
     "prop-types": "^15.6.0",
     "react": "^16.2.0",
     "react-bootstrap": "^0.31.5",
+    "react-debounce-input": "^3.2.0",
```
Comment (ohadlevy) [inline package.json:80]: @ekohl ack/nack?

Comment (ekohl) [reply package.json:80]: Because we branched very late this was already included so :+1: from a packaging perspective

**Review by ekohl — APPROVED** [review]
Packaging should be good.

**Comment by ohadlevy** [comment]
there are 0 failures on the katello test at https://ci.theforeman.org/job/test_develop_pr_katello/7457/ yet its marked as failed, any idea /cc @ehelms  ?

**Review by xprazak2 — APPROVED** [review]
Merging, thanks!

**Pull request merged**